### PR TITLE
Persist tab name in url for schema page

### LINKED ISSUE #1922: Persist schema page's tab name in the url

### PR DESCRIPTION
**Pull request by rajatvijay**
Fixes #1922

**Screenshots**
https://www.loom.com/share/e76803ee521c46d99436a32319609dbf

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: mathesar_ui/src/pages/schema/SchemaPage.svelte
Comment (pavish) [inline mathesar_ui/src/pages/schema/SchemaPage.svelte:107]: The tabs here should not be buttons with an on click event navigating to a route. They should be anchors, so that users can also right click and open them from in a different browser window/tab.

**Review by pavish — CHANGES_REQUESTED** [review]

File: mathesar_ui/src/component-library/tabs/TabContainer.svelte
Comment (pavish) [inline mathesar_ui/src/component-library/tabs/TabContainer.svelte:33]: Why is active tab not set here for tabs with links?

Comment (pavish) [reply mathesar_ui/src/component-library/tabs/TabContainer.svelte:33]: This component is common and this property needs to be handled by the component. Binding to the property from the parent is optional, so it's not a concern if it gets changed internally in this scenario since we control the property on the parent.

**Review by pavish — CHANGES_REQUESTED** [review]
@rajatvijay Looks good!

I do have a comment I'd like to be resolved before merging this in.

File: mathesar_ui/src/component-library/tabs/TabContainer.svelte
Comment (rajatvijay) [reply mathesar_ui/src/component-library/tabs/TabContainer.svelte:33]: @pavish 

This is done to avoid the state mess up created by the two data binding when the user does cmd+click on the tabs with links. 

Here is a video showcasing this: https://www.loom.com/share/e2be4607c15847beac3997c5c43df58c

Comment (pavish) [reply mathesar_ui/src/component-library/tabs/TabContainer.svelte:34]: @rajatvijay The solution made above is a workaround. The proper way to handle this is to select the tab only on the click event (not on the mouse down event), and to check if 'cmd/ctrl' is pressed down when click event is raised and if the tab is a link, and do nothing if it's a cmd+click and leave it to the browser to take over. I believe this is small enough to handle it in this PR itself.

**Review by pavish — CHANGES_REQUESTED** [review]

**Review by pavish — APPROVED** [review]
@rajatvijay I went through the tab component and I'd very much like to deprecate this and write a new one altogether. I'd prefer the new component to be more headless like the ListBox. We can queue this for later, after our release.

For now, I'm gonna go ahead and merge this in, since a new component would lead to rewrite this logic anyway.

**Pull request merged**
